dGame/dCinema/Recorder.cpp

```diff
@@ -0,0 +1,1200 @@
+	}
+
+	if (!timeoutLabel.empty()) {
+		element->SetAttribute("timeoutLabel", timeoutLabel.c_str());
+	}
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::BarrierRecord::Deserialize(tinyxml2::XMLElement* element) {
+	signal = element->Attribute("signal");
+	
+	if (element->Attribute("timeout")) {
+		timeout = element->FloatAttribute("timeout");
+	}
+
+	if (element->Attribute("timeoutLabel")) {
+		timeoutLabel = element->Attribute("timeoutLabel");
+	}
+
+	m_Delay = element->DoubleAttribute("t");
+}
+
+Cinema::Recording::SignalRecord::SignalRecord(const std::string& signal) {
+	this->signal = signal;
+}
+
+void Cinema::Recording::SignalRecord::Act(Entity* actor) {
+}
+
+void Cinema::Recording::SignalRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("SignalRecord");
+
+	element->SetAttribute("signal", signal.c_str());
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::SignalRecord::Deserialize(tinyxml2::XMLElement* element) {
+	signal = element->Attribute("signal");
+
+	m_Delay = element->DoubleAttribute("t");
+}
+
+Cinema::Recording::PlayerProximityRecord::PlayerProximityRecord(float distance, float timeout, const std::string& timeoutLabel) {
+	this->distance = distance;
+	this->timeout = timeout;
+	this->timeoutLabel = timeoutLabel;
+}
+
+void Cinema::Recording::PlayerProximityRecord::Act(Entity* actor) {
+}
+
+void Cinema::Recording::PlayerProximityRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("PlayerProximityRecord");
+
+	element->SetAttribute("distance", distance);
+	if (timeout > 0.0f) {
+		element->SetAttribute("timeout", timeout);
+	}
+
+	if (!timeoutLabel.empty()) {
+		element->SetAttribute("timeoutLabel", timeoutLabel.c_str());
+	}
+
+	element->SetAttribute("t", m_Delay);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::PlayerProximityRecord::Deserialize(tinyxml2::XMLElement* element) {
+	distance = element->FloatAttribute("distance");
+
+	if (element->Attribute("timeout")) {
+		timeout = element->FloatAttribute("timeout");
+	}
+
+	if (element->Attribute("timeoutLabel")) {
+		timeoutLabel = element->Attribute("timeoutLabel");
+	}
+
+	m_Delay = element->DoubleAttribute("t");
+}
+
+void Cinema::Recording::ConcludeRecord::Act(Entity* actor) {
+}
+
+void Cinema::Recording::ConcludeRecord::Serialize(tinyxml2::XMLDocument& document, tinyxml2::XMLElement* parent) {
+	auto* element = document.NewElement("ConcludeRecord");
+
+	element->SetAttribute("t", m_Delay);
+
+	element->SetAttribute("cleanUp", true);
+
+	parent->InsertEndChild(element);
+}
+
+void Cinema::Recording::ConcludeRecord::Deserialize(tinyxml2::XMLElement* element) {
+	m_Delay = element->DoubleAttribute("t");
+
+	if (element->Attribute("cleanUp")) {
+		cleanUp = element->BoolAttribute("clean-up");
```

Comment:
```suggestion
		cleanUp = element->BoolAttribute("cleanUp");
```

Reply:
also as a note, bool attribute already does bounds checks for you, the second argument allows you to choose a default value should one not exist in the xml doc